app/validators/form_password_validator.rb

```diff
@@ -2,8 +2,20 @@ module FormPasswordValidator
   extend ActiveSupport::Concern
 
   included do
+    attr_accessor :password
+    attr_reader :user
+
     validates :password,
               presence: true,
               length: Devise.password_length
+
+    validate :valid_password
+  end
+
+  def valid_password
+    return if errors[:password].any?
+    temp_user = User.new(password: password)
+    return if temp_user.valid?
+    errors.add :password, temp_user.errors[:password]
```

Comment:
With `min_password_score` set to `3`, the gem did not treat a 7-char password as weak. With a setting of `4`, I tried a bunch of 7 character passwords, and they were all considered too weak. So, if we use the default and recommended setting, it sounds like we can eliminate the length validation.

Also, note that `temp_user.errors[:password]` is an array, so adding it to the form's errors will cause the error message displayed to the user to include brackets. To avoid that, add `[0]` at the end of the line.